Library/Homebrew/brew.sh

```diff
@@ -419,10 +419,23 @@ update-preinstall() {
   [[ -z "$HOMEBREW_AUTO_UPDATE_CHECKED" ]] || return
   [[ -z "$HOMEBREW_UPDATE_PREINSTALL" ]] || return
 
-  if [[ "$HOMEBREW_COMMAND" = "install" || "$HOMEBREW_COMMAND" = "upgrade" ||
+  if [[ "$HOMEBREW_COMMAND" = "cask" ]]
+  then
+    if [[ "$HOMEBREW_CASK_COMMAND" != "upgrade" && $HOMEBREW_ARG_COUNT -lt 3 ]]
+    then
+      return
+    fi
+  elif [[ "$HOMEBREW_COMMAND" != "upgrade" && $HOMEBREW_ARG_COUNT -lt 2 ]]
+  then
+    return
```

Comment:
the only behavior difference I can see is that the return statements preventing the `export HOMEBREW_AUTO_UPDATE_CHECKED="1"` from being set at the end of this function, but I don't know why that would make a difference

Reply:
This was it, well done @scpeters 👏

Reply:
:grin: